### PR TITLE
Fix HTTP tests and test bad tarballs are rejected

### PR DESCRIPTION
**Pull request by alexcrichton**
Continuation of #1054

### CODE REVIEW

File: src/tests/record.rs [hunk 1]
```diff
@@ -69,221 +87,219 @@ fn cache_file(name: &str) -> PathBuf {
...
         }
-        panic!("didn't receive header: {}", line);
+        panic!("didn't find header {:?}", (name, value));
     }
+    let req_body = exchange.request.body;
+    let verify_body = req.body().concat2().map(move |body| {
+        assert_eq!(&body[..], &req_body[..]);
+    });
 
-    // TODO: validate the body
```
Comment (carols10cents) [inline src/tests/record.rs]: :+1:!

**Comment by carols10cents** [comment]
bors: r+

**Comment by alexcrichton** [comment]
I've removed the bufstream dependency now

**Comment by carols10cents** [comment]
bors: r+

**Pull request merged**